### PR TITLE
Style source input states

### PR DESCRIPTION
**Pull request by kof**
## Description

1. Styling all states
Screenshot 2023-01-14 at 15 40 54

2. Functionality overview

- Type a new input with autocomplete
- Select an existing source from a list
- Enter a new source
- Hover the source to see the menu
- Menu provides: Remove, Duplicate, Disable, Edit name
- Drag and drop to reorder
- Click to toggle select/unselect
- Double click to edit name
- Local source can only be disabled, nothing else should be possible
- Hit Backspace to delete the last Source item when you are in the input

https://user-images.githubusercontent.com/52824/212717241-eb694803-ee50-4054-bdd6-faaa6ffb197b.mp4

## Steps for reproduction

1. open stories in the designer
3. check all possible states are according to [figma](https://www.figma.com/file/sfCE7iLS0k25qCxiifQNLE/%F0%9F%93%9A-Webstudio-Library?node-id=947%3A3701&t=GviFHhyBu6zfoxRk-4)

## Code Review

- [ ] hi @rpominov , I need you to do
  - conceptual review (architecture, feature-correctness)
  - check how I exposed buttonStyle from design system

- [ ] hi @taylornowotny 
  - please try in storybook and see if you find mistakes. Autocomplete list/menu is not done, only the style sources.
  
## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx [hunk 1]
```diff
@@ -39,7 +41,75 @@ const menuCssVars = ({ show }: { show: boolean }) => ({
...
+  top: 0,
+  height: "100%",
+  padding: 0,
+  borderTopLeftRadius: 0,
+  borderBottomLeftRadius: 0,
+  visibility: cssVars.use(menuTriggerVisibilityVar),
+  variants: {
+    source: {
+      local: {
+        background: "$colors$backgroundStyleSourceToken",
```
Comment (rpominov) [inline apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx:56]: I think you don't need this background. Nothing changes when I toggle it on and off in browser

Screenshot 2023-01-14 at 20 07 20

Same for other values of `source`

File: apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx [hunk 2]
```diff
@@ -39,7 +41,75 @@ const menuCssVars = ({ show }: { show: boolean }) => ({
   [menuTriggerVisibilityVar]: show ? "visible" : "hidden",
 });
 
+const MenuTrigger = styled("button", baseButtonStyle, {
```
Comment (rpominov) [inline apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx:44]: Not sure it's a good idea to import `baseButtonStyle` from button. Now every time someone needs to make changes to button, they would also have to test whether they broke this component

**Review by rpominov — APPROVED** [review]

File: apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx (hunk 2 above)
Comment (kof) [reply apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx:44]: alright, removed

File: apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx (hunk 1 above)
Comment (kof) [reply apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx]: yep, its the same color as the button, removed

**Pull request merged**